nexus/src/defaults.rs

```diff
@@ -0,0 +1,79 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Default values for data in the Nexus API, when not provided explicitly in a request.
+
+use ipnetwork::Ipv6Network;
+use lazy_static::lazy_static;
+use omicron_common::api::external;
+use omicron_common::api::external::Ipv6Net;
+use std::net::Ipv6Addr;
+
+lazy_static! {
+    pub static ref DEFAULT_FIREWALL_RULES: external::VpcFirewallRuleUpdateParams =
+        serde_json::from_str(r#"{
+            "allow-internal-inbound": {
+                "status": "enabled",
+                "direction": "inbound",
+                "targets": [ { "type": "vpc", "value": "default" } ],
+                "filters": { "hosts": [ { "type": "vpc", "value": "default" } ] },
+                "action": "allow",
+                "priority": 65534,
+                "description": "allow inbound traffic to all instances within the VPC if originated within the VPC"
+            },
+            "allow-ssh": {
+                "status": "enabled",
+                "direction": "inbound",
+                "targets": [ { "type": "vpc", "value": "default" } ],
+                "filters": { "ports": [ "22" ], "protocols": [ "TCP" ] },
+                "action": "allow",
+                "priority": 65534,
+                "description": "allow inbound TCP connections on port 22 from anywhere"
+            },
+            "allow-icmp": {
+                "status": "enabled",
+                "direction": "inbound",
+                "targets": [ { "type": "vpc", "value": "default" } ],
+                "filters": { "protocols": [ "ICMP" ] },
+                "action": "allow",
+                "priority": 65534,
+                "description": "allow inbound ICMP traffic from anywhere"
+            },
+            "allow-rdp": {
+                "status": "enabled",
+                "direction": "inbound",
+                "targets": [ { "type": "vpc", "value": "default" } ],
+                "filters": { "ports": [ "3389" ], "protocols": [ "TCP" ] },
+                "action": "allow",
+                "priority": 65534,
+                "description": "allow inbound TCP connections on port 3389 from anywhere"
+            }
+        }"#).unwrap();
+}
+
+pub fn random_unique_local_ipv6() -> Result<Ipv6Net, external::Error> {
+    use rand::Rng;
+    let mut bytes = [0u16; 8];
+    bytes[0] = 0xfd00;
+    rand::thread_rng().try_fill(&mut bytes[1..4]).map_err(|_| {
+        external::Error::internal_error(
+            "Unable to allocate random IPv6 address range",
+        )
+    })?;
+    Ok(Ipv6Net(Ipv6Network::new(Ipv6Addr::from(bytes), 48).unwrap()))
+}
```

Comment:
maybe this makes no sense, but I find it a little weird to think of a random value as a default value

Reply:
See [RFD 21 section 3.4.2](https://rfd.shared.oxide.computer/rfd/0021#_vpc_apis). Clients can specify a prefix if they wish, or a VPC is assigned a default. This is a [unique local address](https://en.wikipedia.org/wiki/Unique_local_address), defined in [RFC 4193](https://www.ietf.org/archive/id/draft-hain-ipv6-ulac-02.txt). It's just part of the spec.

Reply:
Sorry for being unclear — I was talking about the creation of this file called `defaults.rs` as an organizational nitpick, not to the use of this function. I see now that it exists in order to be able to share common code between `model.rs` and `nexus.rs`. I'm not sure `defaults` will hold up as a category of things that belong together but it's not very important.

Reply:
Ah, I see. Yeah, some of this is shared in a few places. I intended this to be a location for common default values (or ways to generate them) that are part of the public API, but I'm not dogmatic about where it goes.